hudi-common/src/main/java/org/apache/hudi/common/table/timeline/CompletionTimeQueryView.java

```diff
@@ -107,46 +106,13 @@ public boolean isCompletedBefore(String baseInstant, String instantTime) {
    * Returns whether the give instant time {@code instantTime} is sliced after or on the base instant {@code baseInstant}.
    */
   public boolean isSlicedAfterOrOn(String baseInstant, String instantTime) {
-    Option<String> completionTimeOpt = getCompletionTime(baseInstant, instantTime);
+    Option<String> completionTimeOpt = getCompletionTime(instantTime);
     if (completionTimeOpt.isPresent()) {
       return HoodieTimeline.compareTimestamps(completionTimeOpt.get(), GREATER_THAN_OR_EQUALS, baseInstant);
     }
     return true;
   }
 
-  /**
-   * Get completion time with a base instant time as a reference to fix the compatibility.
-   *
-   * @param baseInstant The base instant
-   * @param instantTime The instant time to query the completion time with
-   *
-   * @return Probability fixed completion time.
-   */
-  public Option<String> getCompletionTime(String baseInstant, String instantTime) {
-    Option<String> completionTimeOpt = getCompletionTime(instantTime);
-    if (completionTimeOpt.isPresent()) {
```

Comment:
We can not remove this unless you fix all the test cases that use Integer string as the compaction instant time.

Reply:
yeah, there are too many tests doing that. i've kept this condition just for tests.